src/components/TodoList.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable react/react-in-jsx-scope */
+import TodoNote from "./TodoNote";
+import "../index.css";
+import React, {useState, useEffect, useRef} from "react";
+import {nanoid} from "nanoid";
+import AddNote from "./AddNote";
+
+const TodoList = () => {
+  const getNotesFunction = () => {
+    const savedNotes = localStorage.getItem("react-notes-data");
+    return savedNotes ? JSON.parse(savedNotes) : [];
+  };
+
+  const [notes, setNotes] = useState(getNotesFunction);
+  const draggedOverNote = useRef();
+
+  const dragEnter = (noteToReplace) => {
+    draggedOverNote.note = noteToReplace;
+  };
+
+    const drop = (draggedNote) => {
+      let reorderedNotes = [];
+      let draggedIndex = -1;
```

Comment:
Why do you still need the index if you have the ID?

Think about it logically, rather than index shouldn't you decide how to handle each element based on their ID?

If it is the ID of the draggable, then append the dragged over. If it is the ID of the dragged over, append the draggable. All else append to list.

Reply:
I have fixed this and pushed onto github.